File: forge/templates/flutter-frontend-template/{{project_slug}}/test/src/theme/design_tokens_test.dart
import 'package:{{project_slug}}/src/theme/design_tokens.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  group('DesignTokens spacing', () {
    test('all spacing values are positive', () {
      const spacings = [
        DesignTokens.p4,
        DesignTokens.p8,
        DesignTokens.p12,
        DesignTokens.p16,
        DesignTokens.p20,
        DesignTokens.p24,
        DesignTokens.p32,
        DesignTokens.p48,
        DesignTokens.p64,
      ];
      for (final s in spacings) {
        expect(s, greaterThan(0), reason: 'spacing $s should be positive');
      }
    });

    test('spacing scale increases monotonically', () {
      const spacings = [
        DesignTokens.p4,
        DesignTokens.p8,
        DesignTokens.p12,
        DesignTokens.p16,
        DesignTokens.p20,
        DesignTokens.p24,
        DesignTokens.p32,
        DesignTokens.p48,
        DesignTokens.p64,
      ];
      for (var i = 1; i < spacings.length; i++) {
        expect(
          spacings[i],
          greaterThan(spacings[i - 1]),
          reason: 'spacing[${i}] should be > spacing[${i - 1}]',
        );
      }
    });
  });

  group('DesignTokens border radius', () {
    test('all radius values are positive', () {
      expect(DesignTokens.radiusSmall, greaterThan(0));
      expect(DesignTokens.radiusMedium, greaterThan(0));
      expect(DesignTokens.radiusLarge, greaterThan(0));
      expect(DesignTokens.radiusXLarge, greaterThan(0));
    });

    test('radius scale increases monotonically', () {
      const radii = [
        DesignTokens.radiusSmall,
        DesignTokens.radiusMedium,
        DesignTokens.radiusLarge,
        DesignTokens.radiusXLarge,
      ];
      for (var i = 1; i < radii.length; i++) {
        expect(
          radii[i],
          greaterThan(radii[i - 1]),
          reason: 'radius[${i}] should be > radius[${i - 1}]',
        );
      }
    });
  });

  group('DesignTokens breakpoints', () {
    test('compact is less than or equal to medium', () {
      expect(DesignTokens.compactWidth, lessThanOrEqualTo(DesignTokens.mediumWidth));
    });
  });

  group('DesignTokens icon sizes', () {
    test('icon sizes increase from XS to Hero', () {
      const icons = [
        DesignTokens.iconXS,
        DesignTokens.iconSM,
        DesignTokens.iconMD,
        DesignTokens.iconLG,
        DesignTokens.iconXL,
        DesignTokens.iconHero,
      ];
      for (var i = 1; i < icons.length; i++) {
        expect(
          icons[i],
          greaterThan(icons[i - 1]),
          reason: 'icon[${i}] should be > icon[${i - 1}]',
        );
      }
    });
  });

  group('DesignTokens sidebar', () {
    test('collapsed width is less than expanded width', () {
      expect(
        DesignTokens.sidebarCollapsedWidth,
        lessThan(DesignTokens.sidebarExpandedWidth),
      );
    });

    test('icon column width equals collapsed minus 2x body padding', () {
      expect(
        DesignTokens.sidebarIconColumnWidth,
        DesignTokens.sidebarCollapsedWidth - DesignTokens.sidebarBodyPadding * 2,
      );
    });
  });
}

File: forge/templates/svelte-frontend-template/template/src/lib/shared/lib/design-tokens.test.ts
import { describe, it, expect } from 'vitest';
import { DesignTokens } from '$lib/shared/lib/design-tokens';

describe('DesignTokens', () => {
	it('has positive spacing values in monotonically increasing order', () => {
		const spacings = [
			DesignTokens.p4,
			DesignTokens.p8,
			DesignTokens.p12,
			DesignTokens.p16,
			DesignTokens.p20,
			DesignTokens.p24,
			DesignTokens.p32,
			DesignTokens.p48,
			DesignTokens.p64
		];
		for (let i = 0; i < spacings.length; i++) {
			expect(spacings[i]).toBeGreaterThan(0);
			if (i > 0) {
				expect(spacings[i]).toBeGreaterThan(spacings[i - 1]);
			}
		}
	});

	it('has positive border radius values', () => {
		expect(DesignTokens.radiusSmall).toBeGreaterThan(0);
		expect(DesignTokens.radiusMedium).toBeGreaterThan(0);
		expect(DesignTokens.radiusLarge).toBeGreaterThan(0);
		expect(DesignTokens.radiusXLarge).toBeGreaterThan(0);
		expect(DesignTokens.radiusSmall).toBeLessThan(DesignTokens.radiusMedium);
		expect(DesignTokens.radiusMedium).toBeLessThan(DesignTokens.radiusLarge);
	});

	it('has positive animation durations', () => {
		expect(DesignTokens.durationFast).toBeGreaterThan(0);
		expect(DesignTokens.durationNormal).toBeGreaterThan(0);
		expect(DesignTokens.durationSlow).toBeGreaterThan(0);
		expect(DesignTokens.durationDebounce).toBeGreaterThan(0);
		expect(DesignTokens.durationFast).toBeLessThan(DesignTokens.durationNormal);
		expect(DesignTokens.durationNormal).toBeLessThan(DesignTokens.durationSlow);
	});

	it('has reasonable sidebar layout dimensions', () => {
		expect(DesignTokens.sidebarCollapsedWidth).toBeGreaterThan(0);
		expect(DesignTokens.sidebarExpandedWidth).toBeGreaterThan(DesignTokens.sidebarCollapsedWidth);
	});

	it('has reasonable chat layout dimensions', () => {
		expect(DesignTokens.minChatWidth).toBeGreaterThan(0);
		expect(DesignTokens.maxChatWidth).toBeGreaterThan(DesignTokens.minChatWidth);
		expect(DesignTokens.chatDrawerWidth).toBeGreaterThanOrEqual(DesignTokens.minChatWidth);
		expect(DesignTokens.chatDrawerWidth).toBeLessThanOrEqual(DesignTokens.maxChatWidth);
	});

	it('has positive header and navigation dimensions', () => {
		expect(DesignTokens.headerHeight).toBeGreaterThan(0);
		expect(DesignTokens.bottomNavHeight).toBeGreaterThan(0);
		expect(DesignTokens.minMainAreaWidth).toBeGreaterThan(0);
	});
});
